h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/strategy/AllCategoricalTEApplicationStrategy.java

```diff
@@ -0,0 +1,26 @@
+package ai.h2o.targetencoding.strategy;
+
+import water.fvec.Frame;
+
+import java.util.ArrayList;
+
+public class AllCategoricalTEApplicationStrategy extends TEApplicationStrategy {
+  
+  private Frame _frame;
+  private String _responseColumnName;
+  
+  public AllCategoricalTEApplicationStrategy(Frame frame, String responseColumnName) {
+    _frame = frame; 
+    _responseColumnName = responseColumnName;
+  }
+  
+  public String[] getColumnsToEncode() {
+    ArrayList<String> categoricalColumnNames = new ArrayList<>();
+    for( String vecName : _frame.names()) {
+      if(_frame.vec(vecName).isCategorical() && !vecName.equals(_responseColumnName))
```

Comment:
should we not filter out non-predictors columns like `fold_column`, `weights_column`, `offset_column` as well?

Reply:
well, those can never be categorical... even `fold_column`?

Reply:
from doc:
```
The fold column can include integers (0, 1, 2, … , N-1 values or 1, 2, 3… , N values) or categorical values.
```

Reply:
@seb-h2o I did filtering ( _responseColumnName) but I did not know that fold_column can be categorical. Thanks! I will consider to add `excludeColumnNames` parameter as you suggested because yes... it is already a second column we need to filter out.